src/js/common/components/Challenge/ThanksForViewingChallenge.jsx

```diff
@@ -0,0 +1,130 @@
+import { IconButton } from '@mui/material';
+import { Close } from '@mui/icons-material';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import Confetti from 'react-confetti';
+import React, { useState, useEffect } from 'react';
+
+const ThanksForViewingChallenge = ({ challengeOwner }) => {
+  const [isClosing, setIsClosing] = useState(false);
+  const [showConfetti, setShowConfetti] = useState(false)
+
+  useEffect(() => {
+    if (isClosing) {
+      const timer = setTimeout(() => {
+      }, 500);
+      return () => clearTimeout(timer);
+    }
+  }, [isClosing]);
+
+  useEffect(() => {
+      // Show confetti when the component mounts
```

Comment:
Please run `npm run lint` in the terminal to catch any linting errors. It seems like the indenting here has two extra spaces.